vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gap.c

```diff
@@ -624,32 +624,65 @@ BTStatus_t prvBTSetAdvData( uint8_t ucAdapterIf,
         fields.svc_data_uuid128_len = usServiceDataLen;
     }
 
+    fields.num_uuids16=0;
+    fields.num_uuids32=0;
+    fields.num_uuids128=0;
     if( pxServiceUuid != NULL )
     {
-        if( pxServiceUuid->ucType == eBTuuidType16 )
+      for (size_t i = 0; i < xNbServices; i++)
```

Comment:
The code in this loop contains 3 copies of identical operations that differ only in the type they are applied to:

- Set type
- Set value
- Set UUID
- Increment Number of UUIDs of this type
- Set the is_complete flag for this type of UUID

Please refactor this code so that this pattern is only done once in the loop.  An easy way to do this would be with some small utility functions for the UUID16, UUID32, and UUID128 types that this loop calls based on the value of `pxServiceUuid[i].ucType`.

Reply:
Fair point, that code is not optimal.
However, the structure of the code was there in that form before the PR, it's not code I wrote,
and its structure is not affected by the code I wrote to fix the issue that I reported.


I think refactoring that code is out of scope of this PR, whose initial intent was fixing a bug I found in AFR.